app/src/main/java/com/example/hris/ui/viewmodels/LoginViewModel.kt

```diff
@@ -0,0 +1,42 @@
+package com.example.hris.ui.viewmodels
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import com.example.hris.model.LoginModel
+import com.example.hris.repository.HrisRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class LoginViewModel @Inject constructor(
+    private val hrisRepository: HrisRepository,
+    application: Application
+) : AndroidViewModel(application) {
+
+    val liveDataSuccess = MutableLiveData<Unit>()
+    val loginResponse = MutableLiveData<LoginModel>()
+    val loadingDialogState = MutableLiveData<Boolean>()
+    val message = MutableLiveData<String>()
+
+    fun userLogin(username: String, password: String) {
+        viewModelScope.launch {
+            loadingDialogState.value = true
+            loginResponse.value = hrisRepository.login(username, password)
+            loadingDialogState.value = false
+        }
+    }
+
+    fun isValidLogin() {
```

Comment:
This method is not unnecessary.
1. Saving the `User` object can be done inside `HrisRepository.login()` as it was before.
2. Setting the success or message live data can be done inside the `userLogin()` method.